.github/workflows/odh-notebooks-sync.yml

```diff
@@ -0,0 +1,132 @@
+          directories=()
+          exclude_directories=(
+            ".git/objects/pack"
+            ".github/workflows/",
+          )
+          for path in "${paths[@]}"; do
+            current_dir=$(dirname "$path")
+            #Check if current_dir is not in exclude_directories list
+            if [[ ! "${exclude_directories[@]}" =~ "$current_dir" ]]; then
+              #Check if Pipfile exists in current_dir
+              if [ -f "$current_dir/Pipfile" ];then
+                directories+=("$current_dir")
+              fi
+            fi
+          done
+          # Remove duplicates
+          directories=($(echo "${directories[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' '))
+          # Print the directories for verification
+          echo "Directories (Start updating Pipfile in these below directories in accordance with Codeflare-SDK latest release):"
+          for dir in "${directories[@]}"; do
+            echo "- $dir"
+          done
+          # iterate over the directories and update Pipfile
+          counter=0
+          total=${#directories[@]}
+          for dir in "${directories[@]}"; do
+            counter=$((counter+1))
+            echo -e "\n--Processing directory ($counter)'$dir' of total $total"
+            cd "$dir" && pipenv install codeflare-sdk~="${CODEFLARE_RELEASE_VERSION}" && pipenv --rm && cd -
+            echo "$((total-counter)) directories remaining.."
+          done
+
+      - name: Push changes to forked repository
+        run: |
+          cd $REPO_NAME
+          git add . && git status && git checkout -b ${{ env.UPDATER_BRANCH }} && \
+          git commit -am "Updated Codeflare-SDK via ${{ env.UPDATER_BRANCH }} GitHub action" --signoff  &&
+          git remote set-url origin https://x-access-token:${GITHUB_TOKEN}@github.com/$REPO_OWNER/$REPO_NAME.git
+          git push origin ${{ env.UPDATER_BRANCH }}
+
+      - name: Create Pull Request
+        run: |
+          gh pr create --repo $REPO_OWNER/$REPO_NAME \
+            --title "$pr_title" \
+            --body "$pr_body" \
+            --label "lgtm" \
+            --label "approved" \
+            --head $REPO_OWNER:$UPDATER_BRANCH \
+            --base $BRANCH_NAME
+        env:
+          pr_title: "[Digest Updater Action] Update notebook's pipfile to sync with Codeflare-SDK release"
+          pr_body: |
+            :rocket: This is an automated Pull Request.
+
+            This PR updates the `Pipfile` to sync with latest Codeflare-SDK release.
+
+            :exclamation: **IMPORTANT NOTE**: Remember to delete the ` ${{ env.UPDATER_BRANCH }}` branch after merging the changes
+
+      - name: Wait for PR to merge
+        run: |
+          if git branch -a | grep "${{ env.UPDATER_BRANCH }}"; then
+            timeout 3600 bash -c 'until [[ $(gh pr view '${{ env.UPDATER_BRANCH }}' --json state --jq .state) == "MERGED" ]]; do sleep 5 && echo "$(gh pr view '${{ env.UPDATER_BRANCH }}' --json state --jq .state)"; done'
+          fi
```

Comment:
How is the PR going to be processed in upstream?Is the PR going to be merged automatically? Does it have to wait for someone to approve the PR?

Reply:
Yes, I tried by adding `approved` and `lgtm` labels for auto-merge but not working in this case, so while testing..currently merging it manually